### ISSUE TITLE
type checker gets confused by typealias

### ISSUE THREAD

**Issue by hishamhm** [andremm/typedlua#58]
Test case: 

```lua
typealias Cmd = string|{string}

function f(cmd:Cmd) -- if I change to cmd:string|{string}, it works
   local c: {string} = {}
   if type(cmd) == "string" then
      table.insert(c, cmd)
   else
      c = cmd
   end
end
```

I thought a typealias would be fully equivalent to the type declaration it is aliasing to. Is this a bug or is it by design?

**Comment by andremm** [comment]
> I thought a typealias would be fully equivalent to the type declaration it is aliasing to. Is this a bug or is it by design?

Sorry. This is a bug. Thanks for the report.

**Comment by mascarenhas** [comment]
closed by commit 3df4063d83a9ab502cd1be856a9c6cddc7bf0d74
